src/main/java/sps1101/servlet/AnimalPagesServlet.java

```diff
@@ -25,22 +25,22 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import com.google.sps1101.data.Post;
+import sps1101.data.Post;
 import java.util.ArrayList;
 import java.util.List;
 import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/animal-data")
-public class DataServlet extends HttpServlet {
+public class AnimalPagesServlet extends HttpServlet {
 
     private String page;
-    private List<Post> posts = new ArrayList<>();
+    private List<Post> posts;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     posts = new ArrayList<>();
-    String url;
+    String url = null;
```

Comment:
I don't think this is necessary. I'm fine with leaving it if you think it increases the clarity but `null` will be the default value of any object.